WickedEngine/wiHelper.cpp

```diff
@@ -382,7 +382,11 @@ namespace wiHelper
 	bool FileRead(const std::string& fileName, std::vector<uint8_t>& data)
 	{
 #ifndef PLATFORM_UWP
-		ifstream file(fileName, ios::binary | ios::ate);
+		std::string filepath = fileName;
```

Comment:
There is no need to make a copy of fileName for the windows version, which doesn't modify it. So I preferred your very first solution actually.